Cron task support for Noop Local

derekwolpert opened the pull request:
This PR adds cron support for Noop Local. Enhanced visibility into cron execution will be added to the Noop Local Inspector in an upcoming PR.

Here is an example for how terminal output was adjusted to accommodate cron info:

```
Starting 'server' container
Setting cron schedule for task 'cronTask'
Your application dev server is running on https://localnoop.app:1234
You can control the dev sever with the following keyboard commands:
    [ i ] -> Open the inspector
    [ o ] -> Open the dev server
    [ r ] -> Reload the dev server
    [ q ] -> Stop the dev server
6:21 PM - Running task 'cronTask' on cron schedule
Starting 'cronTask' container
Cron task 'cronTask' completed successfully
6:22 PM - Running task 'cronTask' on cron schedule
Starting 'cronTask' container
Cron task 'cronTask' completed successfully
Stopping local dev server
Removed cron schedule for 'cronTask' container
Stopped 'server' container
```

derekwolpert commented:
@arbitrarytech Removed cron validation from this PR, and added cron validation to `noop-discovery`: https://github.com/noop-cloud/noop-discovery/pull/10

`noop-discovery` PR will need to be merged and released - and the version of `noop-discovery` used by `noop-local` will need to be updated - before this PR is released.

derekwolpert commented:
Latest commit - Added back cron validation, and implemented manual task triggering from the inspector.

arbitrarytech commented on lib/devServer.js:
```diff
@@ -164,13 +164,46 @@ class DevServer {
+    console.log(`Manually Running task '${container.friendlyName}' - start: ${time})`)
+    container.once('end', (statusCode) => {
+      if (statusCode) {
+        console.log(chalk.red(`Error running task '${container.friendlyName}' - start: ${time}, end: ${moment().format('HH:mm:ss')}`))
```
Return?

arbitrarytech reviewed (changes requested):
Looks good. Minor changes.

derekwolpert commented:
@arbitrarytech Issues should be resolved. Removed the unneeded async method, and improved cron functionality.

Also fixed a bug where tasks without LIFECYCLE methods cause an error in Noop Local.
